src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -73,342 +72,318 @@ const defaultProps = {
-                if (this.props.iou.id) {
-                    Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
-                }
-                return;
-            }
-        }
-
-        const prevCurrencyParam = lodashGet(prevProps.route.params, 'currency', '');
-        const currencyParam = lodashGet(this.props.route.params, 'currency', '');
-        if (currencyParam !== '' && prevCurrencyParam !== currencyParam) {
-            this.setState({selectedCurrencyCode: currencyParam});
-        }
-
-        if (prevProps.iou.currency !== this.props.iou.currency) {
-            this.setState({selectedCurrencyCode: this.props.iou.currency});
-        }
+    /*
+      Return the sum of leading zeroes length and absolute amount length(including fraction digits).
+      When the absolute amount is 0, add 2 to the leading zeroes length to represent fraction digits.
+    */
+    return leadingZeroesLength + (absAmount === '0' ? 2 : absAmount.length);
+};
 
-        if (prevProps.iou.amount !== this.props.iou.amount) {
-            const selectedAmountAsString = this.props.iou.amount ? CurrencyUtils.convertToWholeUnit(this.props.iou.currency, this.props.iou.amount).toString() : '';
-            this.setState({
-                amount: selectedAmountAsString,
-                selection: {
-                    start: selectedAmountAsString.length,
-                    end: selectedAmountAsString.length,
-                },
-            });
-        }
-    }
+/**
+ * Check if amount is a decimal up to 3 digits
+ *
+ * @param {String} newAmount
+ * @returns {Boolean}
+ */
+const validateAmount = (newAmount) => {
+    const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
+    return newAmount === '' || (decimalNumberRegex.test(newAmount) && calculateAmountLength(newAmount) <= CONST.IOU.AMOUNT_MAX_LENGTH);
+};
 
-    componentWillUnmount() {
-        if (!this.unsubscribeNavFocus) {
-            return;
-        }
-        this.unsubscribeNavFocus();
-    }
+/**
+ * Replaces each character by calling `convertFn`. If `convertFn` throws an error, then
+ * the original character will be preserved.
+ *
+ * @param {String} text
+ * @param {Function} convertFn - `fromLocaleDigit` or `toLocaleDigit`
+ * @returns {String}
+ */
+const replaceAllDigits = (text, convertFn) =>
+    _.chain([...text])
+        .map((char) => {
+            try {
+                return convertFn(char);
+            } catch {
+                return char;
+            }
+        })
+        .join('')
+        .value();
+
+function MoneyRequestAmountPage(props) {
+    const {translate, toLocaleDigit, fromLocaleDigit, numberFormat} = useLocalize();
+    const selectedAmountAsString = props.iou.amount ? CurrencyUtils.convertToWholeUnit(props.iou.currency, props.iou.amount).toString() : '';
+
+    const prevMoneyRequestId = useRef(lodashGet(props.iou, 'id', ''));
+    const prevProps = useRef(props);
```

Comment:
We don't need to keep the whole props; let's only use `useRef` for the values that we actually need, seems to be just `route.params.currency` and `iou.currency` (add those in two different `useRef` variables)